### PR TITLE
Select GWLF-E in UI

### PR DESCRIPTION
**Pull request by lewfish**
The user can now select the model package (TR-55 or GWLF-E) for a project. The dropdown will be replaced by a better one after the prototype is finished. 

To test, try with `disabled` set to both `true` and `false` [here](https://github.com/lewfish/model-my-watershed/blob/feature/select-gwlfe/src/mmw/js/src/modeling/models.js#L27). TR-55 should work as before, and GWLF-E should poll the gwlfe endpoint, which will return empty results, leading to blank Runoff and Water Quality views and error messages in the console. Hitting back in the model view should work as before. Also, test while logged in.

screen shot 2016-04-01 at 1 47 37 pm 2

Connects #1200

### CODE REVIEW

File: src/mmw/apps/home/urls.py [hunk 1]
```diff
@@ -12,8 +12,9 @@
     url(r'^$', home_page, name='home_page'),
     url(r'^projects/$', projects, name='projects'),
     url(r'^project/$', project, name='project'),
+    url(r'^project/new/', project, name='project'),
     url(r'^project/(?P<proj_id>[0-9]+)/$', project, name='project'),
-    url(r'^project/(?P<proj_id>[0-9]+)/clone/?$',
+    url(r'^project/(?P<proj_id>[0-9]+)/clone/?$~',
```
Comment (kdeloach) [inline src/mmw/apps/home/urls.py]: What does `~` do?

Comment (lewfish) [inline src/mmw/apps/home/urls.py]: Good catch. That's a typo.

**Comment by lewfish** [comment]
Hold of on testing this for now. I found a bug when testing it while logged in that I need to fix.

**Comment by mmcfarland** [comment]
Can you add a flag in the settings for whether to enable the GWLFE option in that list (Similar to the erstwhile StreamSlider)?  It could additionally have "(Coming Soon)" added to the disabled text.  Also I think, even to have this displayed temporarily, we should use the actual model names TR55 = "Site Storm Model", GWLF = "Watershed Multi-year Model"

We'll want to be able to merge into `develop` and `release` without maintaining a long running branch for this in-progress code.

**Comment by lewfish** [comment]
@azaveaci rebuild

**Comment by lewfish** [comment]
@azaveabot rebuild

**Comment by lewfish** [comment]
I updated the commits according to the above feedback, and got things to work when logged in.

File: src/mmw/js/src/analyze/templates/resultsWindow.html [hunk 2]
```diff
@@ -7,18 +7,36 @@
...
+                        {% endif %}
+                    </li>
+                {% endfor %}
+              </ul>
+            </div>
         </li>
     </ul>
 </div>
 
+
```
Comment (rajadain) [inline src/mmw/js/src/analyze/templates/resultsWindow.html]: Is this extra newline intentional?

File: src/mmw/js/src/analyze/templates/resultsWindow.html [hunk 3]
```diff
@@ -7,18 +7,36 @@
...
+              <ul class="dropdown-menu dropdown-menu-right">
+                {% for modelPackage in modelPackages %}
+                    <li>
+                        {% if modelPackage.disabled %}
+                            <a class="disabled" href="#">
+                                {{ modelPackage.displayName }} (Coming Soon)
+                            </a>
+                        {% else %}
+                                <a href="#" data-url="/project/new/{{ modelPackage.name }}">
+                                    {{ modelPackage.displayName }}
```
Comment (rajadain) [inline src/mmw/js/src/analyze/templates/resultsWindow.html]: Inconsistent indentation

File: src/mmw/apps/home/urls.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
     url(r'^$', home_page, name='home_page'),
     url(r'^projects/$', projects, name='projects'),
     url(r'^project/$', project, name='project'),
+    url(r'^project/new/', project, name='project'),
```
Comment (rajadain) [inline src/mmw/apps/home/urls.py]: Is this just a fallback, since there is no actual functionality specific to `/project/new` in the back-end?

Comment (lewfish) [inline src/mmw/apps/home/urls.py]: Yes, it just redirects to /.

File: src/mmw/js/src/modeling/controllers.js [hunk 5]
```diff
@@ -139,37 +133,31 @@ var ModelingController = {
...
 
-                lock.done(function() {
-                    project.on('change:id', updateUrl);
-                    initScenarioEvents(project);
-                    initViews(project);
-                    if (App.projectNumber) {
-                        updateUrl();
-                    }
-                });
+                finishProjectSetup(lock, project);
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: In some functions `lock` is the first argument, and in others it is the second. Consider following a consistent convention.

File: src/mmw/js/src/modeling/controllers.js [hunk 6]
```diff
@@ -139,37 +133,31 @@ var ModelingController = {
...
-            // arrow buttons (see issue #690).
-            App.projectNumber = scenarios.at(0).get('project');
-        }
+        projectCleanUp();
+    },
 
-        App.getMapView().updateModifications(null);
-        App.rootView.subHeaderRegion.empty();
-        App.rootView.sidebarRegion.empty();
+    makeNewProjectCleanUp: function() {
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: Where is this used?

Comment (lewfish) [inline src/mmw/js/src/modeling/controllers.js]: It gets called by the router when exiting the route.

Comment (lewfish) [inline src/mmw/js/src/modeling/controllers.js]: The function name is constructed programmatically, so it doesn't show up if you search the source code.

Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: Ah, magic

**Comment by rajadain** [comment]
Behaviorally works pretty great. Going back and forth creates a new project every time I pass "Model", which is the right behavior for most cases.

For ITSI, passing "Model" should _not_ create a new project, but delete and replace the existing one. This may be complex and warrant a card of its own. I think the behavior should be:

 * If the user has created a project using Model A, goes back to Analyze, and then clicks on Model A again, they should return to existing project.
 * If the user has created a project using Model A, goes back to Analyze, and then clicks on Model B, they should be given a warning that this will kill their existing work, which is in Model A, and they should choose between preserving the project and proceeding with Model A, or killing it and proceeding with Model B.
 * If the user has created a project using Model A, goes back to Analyze, goes back to Draw, and changes their AoI, then they should be given a warning that their project will be killed, and if they agree, we kill the project and let them proceed with whichever model they like.

File: src/mmw/js/src/modeling/models.js [hunk 7]
```diff
@@ -10,6 +10,24 @@ var Backbone = require('../../shim/backbone'),
...
+    {
+        name: TR55_PACKAGE,
+        displayName: 'Site Storm Model',
+        description: ''
+    },
+    {
+        name: GWLFE,
+        displayName: 'Watershed Multi-Year Model',
+        description: '',
+        disabled: true
```
Comment (rajadain) [inline src/mmw/js/src/modeling/models.js]: Consider making this a setting somewhere in [`base.py`](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/mmw/settings/base.py), and threading this through via [`window.clientSettings`](https://github.com/WikiWatershed/model-my-watershed/blob/e1a86e60f3bbc0ae26433f891437414129c281cd/src/mmw/apps/core/templates/base.html#L60).

**Comment by rajadain** [comment]
Other than the comments above, the code looks good and the behavior makes sense. +1!

**Comment by rajadain** [comment]
I would recommend splitting out the ITSI stuff into its own card to be tackled later.

File: src/mmw/js/src/modeling/models.js (hunk 7 above)
Comment (lewfish) [inline src/mmw/js/src/modeling/models.js]: I thought about doing this, but the backend doesn't need access to this list of allowed packages, and I would have to define the constants again on the frontend.

Comment (rajadain) [inline src/mmw/js/src/modeling/models.js]: True, but if this is acting as a feature flag, which we turn on for development but off for staging / production deploys, it makes sense to keep it with all other feature flag-like settings, so there's one place for everything.

**Comment by lewfish** [comment]
Thanks for the review. I pushed a new commit to address your feedback.

**Comment by rajadain** [comment]
+1, sweet fixes!

**Comment by rajadain** [comment]
I think this is causing the tests to fail:

```
$ ./scripts/npm.sh run lint
+ ARGS='run lint'
+ vagrant ssh app -c 'cd /opt/app && envdir /etc/mmw.d/env npm run lint'

> model-my-watershed@0.0.1 lint /opt/app
> jshint $(find js -name *.js -not -path '*/shim/*')

js/src/analyze/views.js: line 10, col 5, 'modelingModels' is defined but never used.

1 error
```

**Comment by lewfish** [comment]
@azavea-bot rebuild

**Comment by lewfish** [comment]
See https://github.com/WikiWatershed/model-my-watershed/issues/1222

**Comment by rajadain** [comment]
:+1:

**Pull request merged**